### ISSUE TITLE
Admin panel to restrict based on members of the context

### ISSUE THREAD

**Issue by techsmyth** [cherrytwist/Client.Web#304]
## Description

As a admin I only want to be able to have users presented to me for adding to groups that are eligible to be added

## Acceptance criteria

- [x] Adding users to groups other than members on a challenge should be restricted to the users in the members group
- [x] Adding users to groups other than members on an opporutnity should be restricted to the users in the members group
- [x] adding users to an opportunity should be restricted to members of the containing challenge
- [x] Adding users to members group at challenge or opportunity level to use the specific mutation for adding members i.e. addUserToOpportunity instead of addUserToGroup for the "members" group of an Opportunity

Basically the members group needs to be handled differently at Challenge and Opportunity level.

## Additional Context
See card #555 for additional context
